build.sbt

```diff
@@ -151,6 +151,24 @@ lazy val repl = project
 //    fork in (Test, testOnly) := true
   )
 
+
+/**
+ * REPL available via remote ssh access. Pluggable into any app environment to allow live hacking on a live application.
+ */
+lazy val sshd = project.
+    dependsOn(repl).
+    settings(sharedSettings:_*).
+    settings(
+      name := "ammonite-repl-sshd",
+      libraryDependencies ++= Seq(
+        "org.apache.sshd" % "sshd-core" % "0.14.0",
+        //-- test --//
+        "org.slf4j" % "slf4j-nop" % "1.7.12" % "test",
+        "com.jcraft" % "jsch" % "0.1.53" % "test",
+        "org.scalacheck" %% "scalacheck" % "1.12.4" % "test"
```

Comment:
ScalaCheck really helped me to find some interesting properties of my initial implementation of auth which I wasn't aware of. I think I would never find them if I would use only predefined testcases. There is no notion of "some incorrect credentials will pass", test clearly states that "for all different credential pairs assert that authentication fails".

In general having ScalaCheck-based tests run against some predefined table is a quite popular pattern, I don't see a problem combining this approaches. ScalaTest even have some special support for such testing.